pkg/asset/agent/image/unconfigured_ignition.go

```diff
@@ -0,0 +1,199 @@
+	if err != nil {
+		return err
+	}
+
+	registriesConfig := &mirror.RegistriesConf{}
+	registryCABundle := &mirror.CaBundle{}
+	dependencies.Get(registriesConfig, registryCABundle)
+
+	infraEnvID := uuid.New().String()
+	logrus.Debug("Generated random infra-env id ", infraEnvID)
+
+	osImage, err := getOSImagesInfo(archName)
+	if err != nil {
+		return err
+	}
+	a.CPUArch = *osImage.CPUArchitecture
+
+	agentTemplateData := &agentTemplateData{
+		PullSecret:                pullSecretAsset.GetPullSecretData(),
+		ReleaseImages:             releaseImageList,
+		ReleaseImage:              clusterImageSet.Spec.ReleaseImage,
+		ReleaseImageMirror:        mirror.GetMirrorFromRelease(clusterImageSet.Spec.ReleaseImage, registriesConfig),
+		HaveMirrorConfig:          len(registriesConfig.MirrorConfig) > 0,
+		PublicContainerRegistries: getPublicContainerRegistries(registriesConfig),
+		InfraEnvID:                infraEnvID,
+		OSImage:                   osImage,
+		Proxy:                     infraEnv.Spec.Proxy,
+	}
+
+	err = bootstrap.AddStorageFiles(&config, "/", "agent/files", agentTemplateData)
+	if err != nil {
+		return err
+	}
+
+	err = addBootstrapScripts(&config, clusterImageSetAsset.Config.Spec.ReleaseImage)
+	if err != nil {
+		return err
+	}
+
+	enabledServices := getDefaultEnabledServices()
+	if len(nmStateConfigs.StaticNetworkConfig) > 0 {
+		err = addStaticNetworkConfig(&config, nmStateConfigs.StaticNetworkConfig)
+		if err != nil {
+			return err
+		}
+
+		enabledServices = append(enabledServices, "pre-network-manager-config.service")
+	}
+
+	ztpManifestsToInclude := [...]asset.File{
+		*infraEnvAsset.File,
+		*clusterImageSetAsset.File,
+		*pullSecretAsset.File,
+	}
+
+	for _, file := range ztpManifestsToInclude {
+		manifestFile := ignition.FileFromBytes(filepath.Join(manifestPath, filepath.Base(file.Filename)),
+			"root", 0600, file.Data)
+		config.Storage.Files = append(config.Storage.Files, manifestFile)
+	}
+
+	err = bootstrap.AddSystemdUnits(&config, "agent/systemd/units", agentTemplateData, enabledServices)
+	if err != nil {
+		return err
+	}
+
+	addMirrorData(&config, registriesConfig, registryCABundle)
+
+	a.Config = &config
+
+	if err := a.generateFile(unconfiguredIgnitionFilename); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// PersistToFile writes the unconfigured ignition in the assets folder.
+func (a *UnconfiguredIgnition) PersistToFile(directory string) error {
+	if a.File == nil {
+		return errors.New("attempting to persist a UnconfiguredIgnition that has not been generated")
+	}
+	unconfiguredIgnFile := filepath.Join(directory, a.File.Filename)
+
+	err := os.WriteFile(unconfiguredIgnFile, a.File.Data, 0o644) //nolint:gosec // no sensitive info
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *UnconfiguredIgnition) generateFile(filename string) error {
+	data, err := ignition.Marshal(a.Config)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal Ignition config")
```

Comment:
nit - error should probably be `failed to Marshal Unconfigured Ignition` (to distinguish it from https://github.com/openshift/installer/blob/master/pkg/asset/ignition/bootstrap/common.go#L217